com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/Distortion/ApplyDistortion.shader

```diff
@@ -12,6 +12,8 @@ Shader "Hidden/HDRP/ApplyDistortion"
         #pragma only_renderers d3d11 playstation xboxone vulkan metal switch
         #pragma editor_sync_compilation
 
+        #pragma multi_compile _ SMOOTH_DISTORTION
```

Comment:
change for a dynamic branch , no need for more variant here given the cost

Reply:
I'd argue that this is a pretty simple shader here. Compilation cost is probably meaningless and maintenance complexity is very low so might as well have the most performant code (even if it's just a tiny bit).

Reply:
I don't think the perf improvement will be visible here, but I think the
 CoreUtils.SetKeyword(context.cmd, "SMOOTH_DISTORTION", data.smoothDistortion);
could easily pollute the global space, so better so keep everything simple for such cases.

Reply:
side note: use saturate (free) and not clamp (min / max ALU) for clamp(distortionBlur, 0.0, 1.0);

Reply:
and if you want to avoid branch at all you can do

float mip = (_ColorPyramidLodCount - 1) * clamp(distortionBlur, 0.0, 1.0);

=> float mip = (_ColorPyramidLodCount - 1) * saturate(distortionBlur) * RoughDistortion;
with RoughDistortion == 0 if smooth
it cost just an extra mul

Reply:
All done, relaunching the tests locally and pushing